src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -224,16 +227,34 @@ public String getCredentialsId() {
      *
      * @return {@link PersonalAccessToken}
      */
-    public PersonalAccessToken getCredentials() {
+    public PersonalAccessToken getCredentials(AccessControlled context) {
         Jenkins jenkins = Jenkins.get();
-        jenkins.checkPermission(CredentialsProvider.USE_OWN);
-        return StringUtils.isBlank(credentialsId) ? null : CredentialsMatchers.firstOrNull(
-            lookupCredentials(
-                PersonalAccessToken.class,
-                jenkins,
-                ACL.SYSTEM,
-                fromUri(defaultIfBlank(serverUrl, GITLAB_SERVER_URL)).build()
-            ), withId(credentialsId));
+        if (context == null) {
+            jenkins.checkPermission(CredentialsProvider.USE_OWN);
+            return StringUtils.isBlank(credentialsId) ? null : CredentialsMatchers.firstOrNull( lookupCredentials(
+                                                                                                    PersonalAccessToken.class,
+                                                                                                    jenkins,
+                                                                                                    ACL.SYSTEM,
+                                                                                                    fromUri(defaultIfBlank(serverUrl, GITLAB_SERVER_URL)).build()
+                                                                                                ), withId(credentialsId));
+        } else {
+            context.checkPermission(CredentialsProvider.USE_OWN);
```

Comment:
Hi, @mymarche ,

Sorry to bothering you again.
I read your comment and read again the source code.
What I understand, This is this specific line which correct the problem right?
At this line Jenkins, or in your case the context of security, check if the user can configure the pipeline or not?

Strange the permission needed is the Job/Build permission and not Job/Configure but it's another problem.

But I don't understand the if else after. The personal access token(PAT) is stocked at the global level, during the creation of the gitlab server in the administration panel, so the old code should be working to find this PAT. Instead, you try to change the code to find the PAT at folder level too? 
It'll be great I think. The possibility to define the gitlab server instance during the creation of the multi branch pipeline.
 But is it not another feature? And only with this change it can't work right? Because the creation of a gitlab server instance is only defined in the administration panel and not during the job creation. And to be perfect, regarding the gitlab project isolation easily. It should be Project access token and not Personnal access token.
Am i Wrong on the last part or not?

Reply:
Hi, @Turiok !


> At this line Jenkins, or in your case the context of security, check if the user can configure the pipeline or not?
Yes

> Am I wrong on the last part or not?
No, you are right.
I created fix only for searching PAT in this case.
I think, define PAT for gitlab server per folder or per job is correct.
But i don't know how to implement it correctly.